### PR TITLE
Bump python version to 3 11

### PR DESCRIPTION
**Pull request by shtlrs**
This updates the python version that the bot runs on to 3.11

### CODE REVIEW

File: .github/workflows/lint-test.yml
Comment (shtlrs) [inline .github/workflows/lint-test.yml:37]: Actually, we might need to preserve the dev input we provide, I'll confim in a bit

**Comment by wookie184** [comment]
The test failure looks interesting, seems to be here:
https://github.com/python-discord/bot/blob/395e058c52ebc718f7c0c9acec710c71f45d7b38/tests/_autospec.py#L53-L54
not surprising since it seems like that was intended to be a private method.

Maybe replacing it with `pkgutil.resolve_name` will fix it? (https://github.com/python/cpython/pull/18544)

**Comment by shtlrs** [comment]
> The test failure looks interesting, seems to be here:
> 
> https://github.com/python-discord/bot/blob/395e058c52ebc718f7c0c9acec710c71f45d7b38/tests/_autospec.py#L53-L54
> 
> 
> not surprising since it seems like that was intended to be a private method.
> Maybe replacing it with `pkgutil.resolve_name` will fix it? ([python/cpython#18544](https://github.com/python/cpython/pull/18544))

I had actually forgotten about this PR.
I'll try to look into the option you proposed.

one other option we had discussed the other day was monkey patching it by including it in our testing utils.

We'd likely want to avoid that if there's another way that can do this out of the box.

**Comment by MarkKoz** [comment]
> The test failure looks interesting, seems to be here:
> 
> https://github.com/python-discord/bot/blob/395e058c52ebc718f7c0c9acec710c71f45d7b38/tests/_autospec.py#L53-L54
> 
> not surprising since it seems like that was intended to be a private method.
> 
> Maybe replacing it with `pkgutil.resolve_name` will fix it? ([python/cpython#18544](https://github.com/python/cpython/pull/18544))

Yeah, this is intended to be like [this](https://github.com/python/cpython/blob/3.11/Lib/unittest/mock.py#L1662-L1665). That's where I got the original code from, and it used to use that internal method.

**Review by MarkKoz — APPROVED** [review]

**Review by wookie184 — APPROVED** [review]
LGTM, tested and seems to work.

Two test warnings are raised due to this:
```
bot-hu2rt1Uv-py3.11\Lib\site-packages\feedparser\encodings.py:29: DeprecationWarning: 'cgi' is deprecated and slated for removal in Python 3.13
    import cgi
```
The issue tracking that is https://github.com/kurtmckee/feedparser/issues/330, it seems it was already fixed (https://github.com/kurtmckee/feedparser/commit/ed8c762286aa1d8a248a6900190bd02c01ac2f8d) and is just waiting on a release.
```
bot-hu2rt1Uv-py3.11\Lib\site-packages\discord\player.py:28: DeprecationWarning: 'audioop' is deprecated and slated for removal in Python 3.13
    import audioop
```
The usage seems to be here: https://github.com/Rapptz/discord.py/blob/bb7668f8a58ba4b8161edeb77f8936ff807d6537/discord/player.py#L632

Given neither of them are issues currently I'm not sure they need to block this. I'll let you merge since you may need to relock first.

**Review by jchristgit — APPROVED** [review]

**Pull request merged**
